src/altcha_clj/core.cljc

```diff
@@ -89,27 +89,30 @@
 
 (defn calculate-expiration-offset
  "Adds `offset-secs` * 1000 to the `start-ts-ms` timestamp value"
-  [start-ts-ms offset-secs from-now?]
-  (if from-now?
-    (+ (* 1000 offset-secs) (now))
-    (+ (* 1000 offset-secs) start-ts-ms))
-  )
+  [start-ts-ms offset-secs]
+    (+ (* 1000 (parse-int offset-secs)) start-ts-ms))
+  
 
 (defn create-challenge 
   "Creates a challenge for the client to solve.
   options is a map of the following keys: 
   - `:algorithm` - algorithm for creating a digest of the challenge, default is **SHA-256**.
      For ClojureScript, it will always be SHA-256.
      Can also be **SHA-1** or **SHA-512**
-  - `:max-number` - highest random number used for generating the challenge. Default is 1e6
-  - `:salt-len` - length of the salt. Default is 12
-  - `:expires` - optional, recommended. Expiration time of the challenge validity in seconds.
-  The value for this parameter is the **offset from current time**, **not** a precalculated value
-  of an UNIX timestamp. This function will, however, assign a timestamp value 
-  to this key in the resulting map, as a result of calculating  
-  `(+ (* 1000 e) (current-time-ms))`  
-  by calling `calculate-expiration-offset`
+  - `:max-number` - highest random number used for generating the challenge. Default is 1e6, represented as a fixed point integer.
+  - `:salt-len` - length of the salt. Default is 12. Longer salts are more computationally expensive.
+  - `:expires` - optional. This value will be implicitly bound if `ttl` 
+  is present 
+  by calling `calculate-expiration-offset`. See below.
+  - `ttl` - Time-to-live in seconds. Needed for calculating challenge expiration time.
```

Comment:
TODO: consider removing this
adding `created-at` and reference time should be enough to remove the risk of miscalculation or mutating `expires`